spec/ParseQuery.Aggregate.spec.js

```diff
@@ -0,0 +1,412 @@
+  const obj1 = new TestObject(data1);
+  const obj2 = new TestObject(data2);
+  const obj3 = new TestObject(data3);
+  const obj4 = new TestObject(data4);
+  return Parse.Object.saveAll([obj1, obj2, obj3, obj4]);
+}
+
+describe('Parse.Query Aggregate testing', () => {
+  beforeEach((done) => {
+    loadTestData().then(done, done);
+  });
+
+  it('should only query aggregate with master key', (done) => {
+    Parse._request('GET', `aggregate/someClass`, {})
+      .then(() => {}, (error) => {
+        expect(error.message).toEqual('unauthorized: master key is required');
+        done();
+      });
+  });
+
+  it('invalid query invalid key', (done) => {
+    const options = Object.assign({}, masterKeyOptions, {
+      body: {
+        unknown: {},
+      }
+    });
+    rp.get(Parse.serverURL + '/aggregate/TestObject', options)
+      .catch((error) => {
+        expect(error.error.code).toEqual(Parse.Error.INVALID_QUERY);
+        done();
+      });
+  });
+
+  it('invalid query group _id', (done) => {
+    const options = Object.assign({}, masterKeyOptions, {
+      body: {
+        group: { _id: null },
+      }
+    });
+    rp.get(Parse.serverURL + '/aggregate/TestObject', options)
+      .catch((error) => {
+        expect(error.error.code).toEqual(Parse.Error.INVALID_QUERY);
+        done();
+      });
+  });
+
+  it('invalid query group objectId required', (done) => {
+    const options = Object.assign({}, masterKeyOptions, {
+      body: {
+        group: {},
+      }
+    });
+    rp.get(Parse.serverURL + '/aggregate/TestObject', options)
+      .catch((error) => {
+        expect(error.error.code).toEqual(Parse.Error.INVALID_QUERY);
+        done();
+      });
+  });
+
+  it('group by field', (done) => {
+    const options = Object.assign({}, masterKeyOptions, {
+      body: {
+        group: { objectId: '$name' },
+      }
+    });
+    rp.get(Parse.serverURL + '/aggregate/TestObject', options)
+      .then((resp) => {
+        expect(resp.results.length).toBe(3);
+        expect(resp.results[0].hasOwnProperty('objectId')).toBe(true);
+        expect(resp.results[1].hasOwnProperty('objectId')).toBe(true);
+        expect(resp.results[2].hasOwnProperty('objectId')).toBe(true);
+        expect(resp.results[0].objectId).not.toBe(undefined);
+        expect(resp.results[1].objectId).not.toBe(undefined);
+        expect(resp.results[2].objectId).not.toBe(undefined);
+        done();
+      }).catch(done.fail);
+  });
+
+  it('group sum query', (done) => {
+    const options = Object.assign({}, masterKeyOptions, {
+      body: {
+        group: { objectId: null, total: { $sum: '$score' } },
+      }
+    });
+    rp.get(Parse.serverURL + '/aggregate/TestObject', options)
+      .then((resp) => {
+        expect(resp.results[0].hasOwnProperty('objectId')).toBe(true);
+        expect(resp.results[0].objectId).toBe(null);
+        expect(resp.results[0].total).toBe(50);
```

Comment:
Just curious what happens if one of the `score` values happens to be unset/null? I'm assuming it just omits it but would be nice to know for sure.

Reply:
Just tried. If unset/null it gets ignored